store/store.go

```diff
@@ -554,6 +554,18 @@ type requestOptions struct {
 
 // doRequest does an authenticated request to the store handling a potential macaroon refresh required if needed
 func (s *Store) doRequest(client *http.Client, reqOptions *requestOptions, user *auth.UserState) (*http.Response, error) {
+	// FIXME: find a better place for this?
```

Comment:
just to be clear, as far as I understand client will be reused between different goroutines, so the *only* safe place to set attributes on it (because it doesn't expose a lock) is on creation. Otherwise you're creating races.